ports/nrf/hal/hal_twi.h

```diff
@@ -41,9 +41,17 @@
 
 #define TWI_IRQ_VALUES (const uint32_t[]){SPIM0_SPIS0_TWIM0_TWIS0_SPI0_TWI0_IRQn, \
                                           SPIM1_SPIS1_TWIM1_TWIS1_SPI1_TWI1_IRQn}
-
 #endif
 
+typedef enum
+{
+    HAL_TWI_ERROR_NONE      = 0x00,    /* No error */
+    HAL_TWI_ERROR_ORE       = 0x01,    /* Overrun error. A new byte was received before previous byte got read by software from the RXD register. (Previous data is lost). */
+    HAL_TWI_ERROR_ANACK     = 0x02,    /* NACK received after sending the address. */
+    HAL_TWI_ERROR_DNACK     = 0x04,    /* NACK received after sending a data byte (write '1' to clear). */
+} hal_twi_error_t;
```

Comment:
These can be `TWI_ERRORSRC_OVERRUN_Msk` etc

Reply:
the reason for having these defined is to abstract away the nrf51_bitfields/nrf52_bitfields.h and provide the defines to upper layer. I'll rework these enum's to use the values from the corresponding header files so that when ERRORSRC is returned, upper layer can test what the cause of error was.